packages/editor/src/components/collab-sidebar/index.js

```diff
@@ -0,0 +1,303 @@
+import { useSelect, useDispatch, resolveSelect } from '@wordpress/data';
+import { useState, useEffect, useMemo, useCallback } from '@wordpress/element';
+import { comment as commentIcon } from '@wordpress/icons';
+import { addFilter } from '@wordpress/hooks';
+import { store as noticesStore } from '@wordpress/notices';
+import { store as coreStore } from '@wordpress/core-data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import PluginSidebar from '../plugin-sidebar';
+import { collabSidebarName } from './constants';
+import { Comments } from './comments';
+import { AddComment } from './add-comment';
+import { store as editorStore } from '../../store';
+import AddCommentButton from './commentButton';
+import AddCommentToolbarButton from './commentButtonToolbar';
+
+const isBlockCommentExperimentEnabled =
+	window?.__experimentalEnableBlockComment;
+const modifyBlockCommentAttributes = ( settings, name ) => {
+	if ( name?.includes( 'core/' ) && ! settings.attributes.blockCommentId ) {
+		settings.attributes = {
+			...settings.attributes,
+			blockCommentId: {
+				type: 'number',
+				default: 0,
+			},
+		};
+	}
+
+	return settings;
+};
+
+// Apply the filter to all core blocks
+addFilter(
+	'blocks.registerBlockType',
+	'block-comment/modify-core-block-attributes',
+	modifyBlockCommentAttributes
+);
+
+/**
+ * Renders the Collab sidebar.
+ */
+export default function CollabSidebar() {
+	const { createNotice } = useDispatch( noticesStore );
+	const { saveEntityRecord, deleteEntityRecord } = useDispatch( coreStore );
+	const { getEntityRecords } = resolveSelect( coreStore );
+
+	// eslint-disable-next-line @wordpress/data-no-store-string-literals
+	const { openGeneralSidebar } = useDispatch( 'core/edit-post' );
+	const [ blockCommentID, setBlockCommentID ] = useState( null );
+	const [ showCommentBoard, setShowCommentBoard ] = useState( false );
+
+	const [ threads, setThreads ] = useState( () => [] );
+	const { postId } = useSelect( ( select ) => {
+		return {
+			postId: select( editorStore ).getCurrentPostId(),
+		};
+	}, [] );
+
+	const clientId = useSelect( ( select ) => {
+		const { getSelectedBlockClientId } = select( blockEditorStore );
+		setBlockCommentID(
+			select( blockEditorStore ).getBlock( getSelectedBlockClientId() )
+				?.attributes.blockCommentId
+		);
+		return getSelectedBlockClientId();
+	}, [] );
+
+	// Get the dispatch functions to save the comment and update the block attributes.
+	const { updateBlockAttributes } = useDispatch( blockEditorStore );
+
+	const openCollabBoard = () => {
+		setShowCommentBoard( true );
+		openGeneralSidebar( 'edit-post/collab-sidebar' );
+	};
+
+	// Function to save the comment.
+	const addNewComment = async ( comment, parentCommentId ) => {
+		const args = {
+			post: postId,
+			content: comment,
+			comment_type: 'block_comment',
+			comment_approved: 0,
+		};
+
+		if ( parentCommentId ) {
+			args.parent = parentCommentId;
```

Comment:
If possible, we should avoid direct mutation of objects. Sometimes as we move code around, it will manifest in unexpected places.